File: languages/js/src/types/anchor.ts
import { Anchor as WasmAnchor } from '../bloock_wasm_api';
import { AnchorNetwork as WasmAnchorNetwork } from '../bloock_wasm_api';

export class AnchorNetwork {
  public name: string;
  public state: string;
  public txHash: string;

  constructor(name: string, state: string, txHash: string) {
    this.name = name;
    this.state = state;
    this.txHash = txHash;
  }

  static fromWasm(anchor: WasmAnchorNetwork): AnchorNetwork {
    return new AnchorNetwork(anchor.name, anchor.state, anchor.tx_hash);
  }
}

export class Anchor {
  public id: number;
  public blockRoots: string[];
  public networks: AnchorNetwork[];
  public root: string;
  public status: string;

  constructor(
    id: number,
    blockRoots: string[],
    networks: AnchorNetwork[],
    root: string,
    status: string
  ) {
    this.id = id;
    this.blockRoots = blockRoots;
    this.networks = networks;
    this.root = root;
    this.status = status;
  }

  static fromWasm(anchor: WasmAnchor): Anchor {
    return new Anchor(
      Number(anchor.id),
      anchor.block_roots,
      anchor.networks.map((n: WasmAnchorNetwork) => AnchorNetwork.fromWasm(n)),
      anchor.root,
      anchor.status
    );
  }
}

File: languages/go/types/anchor.go
package types

type Anchor struct {
	id         int       `json:"id"`
	blockRoots []string  `json:"block_roots"`
	networks   []Network `json:"networks"`
	root       string    `json:"root"`
	status     string    `json:"status"`
}

type AnchorParams struct {
	Timeout int
}

func NewAnchor(id int, blockRoots []string, networks []Network, root, status string) Anchor {
	return Anchor{
		id:         id,
		blockRoots: blockRoots,
		networks:   networks,
		root:       root,
		status:     status,
	}
}

func (a Anchor) ID() int {
	return a.id
}

func (a Anchor) BlockRoots() []string {
	return a.blockRoots
}

func (a Anchor) Networks() []Network {
	return a.networks
}

func (a Anchor) Root() string {
	return a.root
}

func (a Anchor) Status() string {
	return a.status
}
